<launch>

    <!-- Gazebo -->
    <include file="$(find-pkg-share gazebo_ros)/launch/gzserver.launch.py">
        <arg name="world" value="$(find-pkg-share turtlebot3_gazebo)/worlds/turtlebot3_world.world"/>
    </include>

    <include file="$(find-pkg-share gazebo_ros)/launch/gzclient.launch.py">
    </include>

    <!-- Multi robot -->
    <include file="$(find-pkg-share gyak9)/launch/ns_robot.launch.py">
        <arg name="namespace" value="robot1"/>
        <arg name="x_pose" value="-2.0"/>
        <arg name="y_pose" value="0.0"/>
    </include>
    <node pkg="tf2_ros" exec="static_transform_publisher" name="map_to_robot1_odom" output="screen"
        args="0.0 0.0 0.0 0.0 0.0 0.0 map robot1/odom"/>

    <include file="$(find-pkg-share gyak9)/launch/ns_robot.launch.py">
        <arg name="namespace" value="robot2"/>
        <arg name="x_pose" value="2.0"/>
        <arg name="y_pose" value="0.0"/>
    </include>
    <node pkg="tf2_ros" exec="static_transform_publisher" name="map_to_robot2_odom" output="screen"
        args="0.0 0.0 0.0 0.0 0.0 0.0 map robot2/odom"/>

    <!-- Node -->
    <node pkg="gyak9" exec="test_scan" name="scan_tf" output="screen">
        <param name="use_sim_time"  value="true"/>
        <param name="map_frame"     value ="map"/>
    </node>

    <!-- RVIZ -->
    <node pkg="rviz2" exec="rviz2" name="rviz2" args="-d $(find-pkg-share gyak9)/rviz/gyak9.rviz"/>

</launch>